example/x64/amd/default/CMakeLists.txt

```diff
@@ -0,0 +1,52 @@
+#
+# Copyright (C) 2020 Assured Information Security, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+add_executable(example_x64_amd_default)
+
+target_include_directories(example_x64_amd_default PRIVATE
+    src
+)
+
+target_sources(example_x64_amd_default PRIVATE
+    src/start.s
+    src/main.cpp
+)
+
+target_link_libraries(example_x64_amd_default PRIVATE
+    bsl
+    pal
+    runtime
+)
+
+target_compile_options(example_x64_amd_default PRIVATE
```

Comment:
Why did you remove these from the public interface of the runtime CMake target, and instead put them here? Shouldn't the example just inherit these?

Reply:
The "-c" and command line warnings are an artifact of the build system. Downstream extensions will almost certainly end up using ExternalProject_Add, in which case these are not needed, and therefore cannot be in the interface. The triple is debatable where it should go. Once we sort out the Clang triple issue, we can figure out where we want these.

Reply:
Ok, I moved the triples to the public interface for now. The "-c" and warnings need to remain private as we don't know how an example will compile itself. This should be fixed